src/primitive-field/primitive-field.component.ts

```diff
@@ -103,4 +103,15 @@ export class PrimitiveFieldComponent extends AbstractFieldComponent implements O
     return this.schema['disabled'] ? -1 : 1;
   }
 
+  get tooltipPosition() {
+    let tooltipPlacement = 'top';
+    let firstElem = document.getElementById('first');
+    if (firstElem) {
+      // The first top-level element will have the same bottom placement for all it's elements.
+      if (firstElem.querySelector(`[id='${this.pathString}']`)) {
```

Comment:
yes this will prevent querying all the time...I am gonna make the changes tomorrow.